utilities/cli/README.md

```diff
@@ -135,6 +135,196 @@ The new CLI introduces development lifecycle features:
+
+#### **Understanding Modes**
+Modes potentially provide typical configurations for:
+- **Different hardware**
+- **Deployment scenarios** (e.g. development vs. production vs. cloud inference)
+- **Data sources** (e.g. live video vs. recorded files vs. distributed streaming)
+
+#### **Using Modes**
+
+Examples:
+
+```bash
+# Discover available modes for an application
+./holohub modes body_pose_estimation
+
+# Run an application in a specific mode
+./holohub run body_pose_estimation dds                    # DDS video streaming mode
+./holohub run endoscopy_tool_tracking aja                 # AJA capture card mode
+./holohub run holochat standalone                         # Local LLM inference mode
+
+# Build for a specific mode (if mode has special build requirements)
+./holohub build endoscopy_tool_tracking aja_overlay
+
+# Default behavior (no mode specified uses 'default' mode or first available)
+./holohub run body_pose_estimation                        # Uses 'default' mode (or first mode if no 'default')
+```
+
+#### **Mode vs. CLI Arguments**
+**Important**: When a mode defines specific options, you cannot override them with CLI arguments.
+This ensures mode integrity and prevents conflicting configurations.
+
+**Conflict Rules:**
+- If mode defines `run.docker_args` → Cannot use `--docker-opts`
+- If mode defines `build.depends` → Cannot use `--build-with`
+- If mode defines `build.docker_args` → Cannot use `--build-args`
+- If mode defines `build.configure_args` → Cannot use `--configure-args`
+
+#### **For Application Developers**
+Applications can define modes in their `metadata.json`. Here's the complete field reference:
+
+##### **Mode Structure**
+Each mode is defined as a named object under the `modes` key:
+
+```json
+{
+  "metadata": {
+    "modes": {
+      "mode_name": {
+        // Mode configuration fields
+      }
+    }
+  }
+}
+```
+
+##### **Supported Fields for Each Mode**
+
+**Required Fields:**
+- **`description`** *(string)*: Human-readable description of what this mode does
+- **`run`** *(object)*: Run configuration (see run command fields below)
+
+**Optional Fields:**
+- **`requirements`** *(array of strings)*: List of dependency IDs required for this mode
+- **`build`** *(object)*: Build configuration (see build configuration fields below)
+
+##### **Build Configuration Fields (`build` object):**
+- **`depends`** *(array of strings)*: List of operators/dependencies to build with this mode
+- **`docker_args`** *(string or array)*: Docker **build** arguments (equivalent to CLI `--build-args`)
+  - Can be a single string: `"--build-arg CUSTOM=value"`
+  - Or an array: `["--build-arg", "CUSTOM=value"]`
+- **`configure_args`** *(array of strings)*: Additional configure arguments (such as cmake flags) to pass to the local build step
```

Comment:
should we be explicit and name this `cmake_configure_args` ? cc @jjomier

Reply:
this is originally from the `run` script https://github.com/nvidia-holoscan/holohub/blob/0d64dd6022aae855d19448043c0c4e5083603e2c/run#L666
and is used in the current `holohub` script. `cmake_configure_args` or `cmake_config` is also good, I don't have preference here. please let me know your suggestions @tbirdso @jjomier

Reply:
thanks, this is updated to `cmake_config` from `configure_args` in the preset modes.